kernel/environ.ml

```diff
@@ -246,6 +246,9 @@ let add_rewrite_rules l env =
     symb_pats = List.fold_left (fun symb_pats (c, r) -> Cmap_env.update c (add c r) symb_pats) env.symb_pats l
   }
 
+let lookup_rewrite_rules cst env =
+  Cmap_env.find cst env.symb_pats
```

Comment:
```suggestion
  Cmap_env.get cst env.symb_pats
```
and remove the Not_found catches in callers

Reply:
I don't think it hurts to be able to check that there are rewrite rules associated to a constant.

Reply:
Look at the body, if it's Symbol then there is an entry in the map (possibly empty list)

Reply:
but do as you like